emscripten.py

```diff
@@ -114,8 +114,9 @@ def update_settings_glue(metadata, DEBUG):
   shared.Settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE = sorted(set(all_funcs).difference(metadata['exports']))
 
   shared.Settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += metadata['externs']
-  # With the wasm backend the set of implemented functions is identical to the set of exports
-  shared.Settings.IMPLEMENTED_FUNCTIONS = [asmjs_mangle(x) for x in metadata['exports']]
+
+  all_exports = metadata['exports'] + list(metadata['namedGlobals'].keys())
+  shared.Settings.IMPLEMENTED_FUNCTIONS = [asmjs_mangle(x) for x in all_exports]
```

Comment:
looks like this duplicates code on lines 317-318. whichever runs first can write to `IMPLEMENTED_FUNCTIONS` and the other can read?

Reply:
Good catch!